inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp

```diff
@@ -812,575 +813,601 @@ void MKLDNNNormalizeL2Node::setPostOps(mkldnn::primitive_attr &attr, bool initWe
+                    out_data_t *dst_data_b_cb_h = dst_data_b + cb * spatial_dims * attrs.blk_size + h * w_blk_dims;
+                    auto arg = jit_normalize_call_args();
+                    arg.src = src_data_b_cb_h;
+                    arg.dst = dst_data_b_cb_h;
+                    arg.fused_factor = static_cast<float*>(&modulo_inv);  // broadcast once
+                    arg.work_amount = static_cast<size_t>(jcp.w);
+                    arg.oc_off = cb * attrs.blk_size * sizeof(float);
+                    (*normalize_kernel)(&arg);
+                });
+            } else {  // across_spatial: false
+                parallel_for2d(jcp.h, jcp.w, [&](size_t ih, size_t iw) {
+                    // modulo
+                    float modulo = 0.0f;
+                    const in_data_t *src_data_bhw = src_data_b + ih * w_blk_dims + iw * attrs.blk_size;
+                    out_data_t *dst_data_bhw = dst_data_b + ih * w_blk_dims + iw * attrs.blk_size;
+                    auto arg = jit_normalize_call_args();
+                    arg.src = src_data_bhw;
+                    arg.modulo = static_cast<float*>(&modulo);
+                    arg.src_stride = attrs.blk_size * spatial_dims * sizeof(in_data_t);
+                    arg.work_amount = jcp.c / attrs.blk_size;  // CB or CB-1
+                    (*normalize_modulo_kernel)(&arg);
+                    // for tail
+                    size_t padding = CB * attrs.blk_size - jcp.c;
+                    if (padding > 0) {
+                        size_t tail = attrs.blk_size - padding;
+                        const in_data_t *src_data_bhw_lastCB = src_data_bhw + (CB - 1) * attrs.blk_size * spatial_dims;
+                        for (size_t c = 0; c < tail; c++) {
+                            modulo += src_data_bhw_lastCB[c] * src_data_bhw_lastCB[c];
+                        }
                     }
-                }
-            });
+
+                    modulo = std::sqrt(modulo);
+                    float modulo_inv = 1.0f / (epsApply(modulo, attrs.epsMode, attrs.eps));
+
+                    // normalize
+                    arg.dst = dst_data_bhw;
+                    arg.fused_factor = static_cast<float*>(&modulo_inv);  // broadcast
+                    arg.work_amount = CB;
+                    arg.oc_off = 0;
+                    (*normalize_kernel)(&arg);
+                });
+            }
         }
     }
-}
 
-template <typename in_data_t, typename out_data_t>
-void MKLDNNNormalizeL2Node::normalize_nhwc(const in_data_t* src_data, out_data_t* dst_data, const SizeVector& dims) {
-    size_t blk_size = 1;  // elt in vmm
-    if (mayiuse(cpu::x64::avx512_common)) {
-        blk_size = 16;
-    } else if (mayiuse(cpu::x64::avx2)) {
-        blk_size = 8;
-    } else if (mayiuse(cpu::x64::sse41)) {
-        blk_size = 4;
-    }
+    jit_normalize_config_params jcp = {};
+    mkldnn::primitive_attr kernel_attrs;
```

Comment:
for `NormalizeL2JitExecutor` used only inside ctor so no need to save as class member